frontend/components/common/CompatibleDevice.tsx

```diff
@@ -26,102 +28,134 @@ export function CompatibleDevice({
       maxModelLines > 0 && variants.length > maxModelLines
          ? [variants.slice(0, maxModelLines), variants.slice(maxModelLines)]
          : [variants, []];
-   const { isOpen, getDisclosureProps, getButtonProps, onToggle } =
-      useDisclosure();
+   const [detailsRef, isOpen, toggle] = useDetails();
    return (
-      <NextLink href={device.deviceUrl} passHref>
-         <chakra.a
-            display="flex"
-            alignItems="start"
-            transition="all 300m"
-            {...otherProps}
+      <Box
+         display="flex"
+         alignItems="start"
+         transition="all 300m"
+         {...(!truncateModels && {
+            onClick: (event) => {
+               toggle();
+               event.preventDefault();
+               event.stopPropagation();
+            },
+            onKeyDown: (event) => {
+               if ('Enter' === event.code) {
+                  toggle();
+                  event.preventDefault();
+                  event.stopPropagation();
+               }
+            },
+         })}
+         {...otherProps}
+      >
+         <NextLink href={device.deviceUrl} passHref>
+            <Link>
+               <Img
+                  src={device.imageUrl}
+                  alt={device.deviceName ?? ''}
+                  w="12"
+                  mr="3"
+                  borderWidth="1px"
+                  borderStyle="solid"
+                  borderColor="gray.300"
+                  borderRadius="base"
+               />
+            </Link>
+         </NextLink>
+         <Flex
+            minH="12"
+            flexDir="column"
+            alignSelf="flex-start"
+            justifyContent="center"
          >
-            <Img
-               src={device.imageUrl}
-               alt={device.deviceName ?? ''}
-               w="12"
-               mr="3"
-               borderWidth="1px"
-               borderStyle="solid"
-               borderColor="gray.300"
-               borderRadius="base"
-            />
+            <NextLink href="#compatibility" passHref>
```

Comment:
I wonder if there's any advantage to using NextLink for an in-page anchor like this.

Reply:
None that I'm aware of, but you caught a typo! That's supposed to be `device.deviceUrl` I think

Reply:
Kept the next link, because sometimes it's a /Device/... link https://github.com/iFixit/react-commerce/pull/1604/commits/b02d16c0a1446e996e4dab46802b6888c1ad0325